### ISSUE TITLE
Model fields and quantifiers

### ISSUE THREAD

**Issue by adilsonv77** [OpenJML/OpenJML#542]
Running the class below (with version 0.8.20), the program waits a lot of time and gives me 
` JML result of numeric operation is out of range of the target type`

Quantifiers are not supported for models ? With Lists also happened the same.

```
public class Test {

	/*@ spec_public @*/
	private int[] vals= new int[10];
	
	/*@
	  public model int countVals;
	  represents countVals = (\sum int i; 0<=i && i<vals.length; vals[i]);
	  @*/
	
	//@requires countVals > 0;
	public void m() {		
	}
	
	public static void main(String[] args) {
		Test t = new Test();
		t.m();
	}
}
```

**Comment by adilsonv77** [comment]
As workaround, I created model method to sum the values.

**Comment by davidcok** [comment]
Yes - sorry - in the translation of the \sum, the loop test was computed outside the loop - so an infinite loop resulted. Easy fix that will be in V0.8.21 — if you need it urgently tell me. Otherwise do your workaround — for RAC, compute the quantifiers in a model method, and not directly in the invariant or other clause.

**Comment by davidcok** [comment]
Fixed in the just released V0.8.21
